src/Utils/ObjectBeanPropertyDescriptor.php

```diff
@@ -253,14 +253,21 @@ public function getJsonSerializeCode(): string
         return $code;
     }
 
-    private function getLazySerializeCode(string $propertyAccess): string
+    public function getLazySerializeCode(string $propertyAccess): string
     {
         $rows = [];
         foreach ($this->getForeignKey()->getUnquotedForeignColumns() as $column) {
             $descriptor = $this->getBeanPropertyDescriptor($column);
-            $indexName = ltrim($descriptor->getVariableName(), '$');
-            $columnGetterName = $descriptor->getGetterName();
-            $rows[] = "'$indexName' => $propertyAccess->$columnGetterName()";
+            if ($descriptor instanceof ScalarReferencePropertyDescriptor) {
+                $descriptor = $descriptor->getReferencedPropertyDescriptor();
```

Comment:
This could return a `ScalarReferencePropertyDescriptor`. We should get the referenced property recursively.